isis/src/control/objs/CsmBundleObservation/CsmBundleObservation.cpp

```diff
@@ -0,0 +1,876 @@
+   *                                                    settings for this CsmBundleObservation
+   */
+  const BundleObservationSolveSettingsQsp CsmBundleObservation::solveSettings() {
+    return BundleObservationSolveSettingsQsp(nullptr);
+  }
+
+
+  /**
+   * Applies the parameter corrections
+   *
+   * @param corrections Vector of corrections to apply
+   *
+   * @throws IException::Unknown "Instrument position is NULL, but position solve option is
+   *                              [not NoPositionFactors]"
+   * @throws IException::Unknown "Instrument position is NULL, but pointing solve option is
+   *                              [not NoPointingFactors]"
+   * @throws IException::Unknown "Unable to apply parameter corrections to CsmBundleObservation."
+   *
+   * @return @b bool Returns true upon successful application of corrections
+   *
+   * @internal
+   *   @todo always returns true?
+   */
+  bool CsmBundleObservation::applyParameterCorrections(LinearAlgebra::Vector corrections) {
+    // Check that the correction vector is the correct size
+    if (corrections.size() != m_paramIndices.size()) {
+      QString msg = "Invalid correction vector passed to observation.";
+      IException(IException::Programmer, msg, _FILEINFO_);
+    }
+
+    // Apply the corrections to the CSM camera
+    CSMCamera *csmCamera = dynamic_cast<CSMCamera*>(front()->camera());
+    for (size_t i = 0; i < corrections.size(); i++) {
+      csmCamera->applyParameterCorrection(m_paramIndices[i], corrections[i]);
+    }
+
+    // Accumulate the total corrections
+    m_corrections += corrections;
+
+    return true;
+  }
+
+
+  /**
+   * Returns the number of total parameters there are for solving
+   *
+   * The total number of parameters is equal to the number of position parameters and number of
+   * pointing parameters
+   *
+   * @return @b int Returns the number of parameters there are
+   */
+  int CsmBundleObservation::numberParameters() {
+    return m_paramIndices.size();
+  }
+
+
+  /**
+ * @brief Creates and returns a formatted QString representing the bundle coefficients and
+ * parameters
+ *
+ * @depricated The function formatBundleOutputString is depricated as of ISIS 3.9
+ * and will be removed in ISIS 4.0
+ *
+ * @param errorPropagation Boolean indicating whether or not to attach more information
+ *     (corrections, sigmas, adjusted sigmas...) to the output QString
+ * @param imageCSV Boolean which is set to true if the function is being
+ *     called from BundleSolutionInfo::outputImagesCSV().  It is set to false by default
+ *     for backwards compatibility.
+ *
+ * @return @b QString Returns a formatted QString representing the CsmBundleObservation
+ *
+ * @internal
+ *   @history 2016-10-26 Ian Humphrey - Default values are now provided for parameters that are
+ *                           not being solved. Fixes #4464.
+ */
+QString CsmBundleObservation::formatBundleOutputString(bool errorPropagation, bool imageCSV) {
```

Comment:
I left these two functions unimplemented until we figure out how we want to report stuff.